### PR TITLE
Lista ulubionych atrakcji - model i MVC 

### PR DESCRIPTION
**Pull request by wkingakos**

### CODE REVIEW

File: src/main/java/sggw/wzim/czasnawypad/db/dto/FavouriteAttractionDTO.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class FavouriteAttractionDTO {
+	private Integer id;
+	private Integer attractionId;
+	private Integer userId;
+}
```
Comment (Zajcw) [inline src/main/java/sggw/wzim/czasnawypad/db/dto/FavouriteAttractionDTO.java:12]: Missing empty line

File: src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	public FavouriteAttractionDTO addFavourite(User user, CreateFavouriteAttractionDTO dto) {
+		Attraction attraction = attractionRepository.findById(dto.getAttractionId())
+			.orElseThrow(() -> new IllegalArgumentException("Attraction not found"));
+		FavouriteAttraction favourite = FavouriteAttraction.builder()
+			.user(user)
+			.attraction(attraction)
+			.build();
+		return toDto(favouriteRepository.save(favourite));	
+	}
```
Comment (Zajcw) [inline src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java:34]: Maybe we should introduce `userRepository` and find user by id from `CreateFavouriteAttractionDTO` object?

File: src/main/java/sggw/wzim/czasnawypad/db/dto/CreateFavouriteAttractionDTO.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+@Data
+public class CreateFavouriteAttractionDTO {
+	@NotNull
+	private Integer attractionId;
+	@NotNull
+	private Integer userId;
+}
```
[the remark this answers is not in the document]
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/db/dto/CreateFavouriteAttractionDTO.java]: Done

File: src/main/java/sggw/wzim/czasnawypad/db/dto/FavouriteAttractionDTO.java (hunk 1 above)
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/db/dto/FavouriteAttractionDTO.java]: Done

File: src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java (hunk 2 above)
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java]: Done

**Comment by Zajcw** [comment]
@wkingakos please rebase your branch onto `main`

File: src/main/java/sggw/wzim/czasnawypad/controller/AttractionFavouriteController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    private final FavouriteAttractionService favouriteService;
+    
+    @GetMapping
+    public List<FavouriteAttractionDTO> getUserFavourites(@AuthenticationPrincipal User user) {
+        return favouriteService.getFavouritesByUser(user);
+    }
+
+    @PostMapping
+    public FavouriteAttractionDTO addFavourite(@AuthenticationPrincipal User user,
```
[the remark this answers is not in the document]
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/controller/AttractionFavouriteController.java]: Done

File: src/main/java/sggw/wzim/czasnawypad/controller/AttractionFavouriteController.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+@RestController
+@RequestMapping("/favourites")
+@RequiredArgsConstructor
+public class AttractionFavouriteController {
+
+    private final FavouriteAttractionService favouriteService;
+    
+    @GetMapping
+    public List<FavouriteAttractionDTO> getUserFavourites(@AuthenticationPrincipal User user) {
```
[the remark this answers is not in the document]
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/controller/AttractionFavouriteController.java]: Done

**Review by calzone-G — DISMISSED** [review]

File: src/main/java/sggw/wzim/czasnawypad/db/FavouriteAttractionRepository.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.List;
+
+@Hidden
+@Repository
+public interface FavouriteAttractionRepository extends JpaRepository<FavouriteAttraction, Integer> {
+    List<FavouriteAttraction> findByUserId(Integer userId);
+    List<FavouriteAttraction> findByAttractionId(Integer attractionId);
+
+
```
Comment (Zajcw) [inline src/main/java/sggw/wzim/czasnawypad/db/FavouriteAttractionRepository.java:17]: Too many empty lines

File: src/main/java/sggw/wzim/czasnawypad/db/dto/CreateFavouriteAttractionDTO.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+@ToString
+@AllArgsConstructor
+@NoArgsConstructor
+public class CreateFavouriteAttractionDTO {
+	@NotNull
+	private Integer attractionId;
+	@NotNull
+	private Integer userId;
+
+
```
Comment (Zajcw) [inline src/main/java/sggw/wzim/czasnawypad/db/dto/CreateFavouriteAttractionDTO.java:20]: Too many empty lines

File: src/main/java/sggw/wzim/czasnawypad/db/dto/FavouriteAttractionDTO.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.Data;
+
+@Data
+@Builder
+public class FavouriteAttractionDTO {
+	private Integer id;
+	private Integer attractionId;
+	private Integer userId;
+
+
```
Comment (Zajcw) [inline src/main/java/sggw/wzim/czasnawypad/db/dto/FavouriteAttractionDTO.java:13]: Too many empty lines

File: src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+	
+    private FavouriteAttractionDTO toDto(FavouriteAttraction favourite) {
+        return FavouriteAttractionDTO.builder()
+            .id(favourite.getId())
+            .attractionId(favourite.getAttraction().getId())
+			.userId(favourite.getUser().getId())
+			.build();
+	}
```
Comment (Zajcw) [inline src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java:46]: Please introduce MapStruct mapper for this

**Review by Zajcw — CHANGES_REQUESTED** [review]

File: src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java (hunk 9 above)
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java]: Done

File: src/main/java/sggw/wzim/czasnawypad/db/FavouriteAttractionRepository.java (hunk 6 above)
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/db/FavouriteAttractionRepository.java]: Done

File: src/main/java/sggw/wzim/czasnawypad/db/dto/CreateFavouriteAttractionDTO.java (hunk 7 above)
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/db/dto/CreateFavouriteAttractionDTO.java]: Done

File: src/main/java/sggw/wzim/czasnawypad/db/dto/FavouriteAttractionDTO.java (hunk 8 above)
Comment (wkingakos) [reply src/main/java/sggw/wzim/czasnawypad/db/dto/FavouriteAttractionDTO.java]: Done

**Review by Zajcw — APPROVED** [review]

**Pull request merged**